EEJT-000A/src/main/java/com/olehshynkarchuk/task2/FileProcessor.java

```diff
@@ -0,0 +1,41 @@
+package com.olehshynkarchuk.task2;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.io.File;
+
+public abstract class FileProcessor {
+
+    protected FileProcessor() {
+    }
+
+    public void searchFiles(File filePath, FileParameters fileParameters) {
+        if (needToSearchCorrespondingFiles(fileParameters)) {
+            walkingOfDirectoryByRecursion(filePath, fileParameters);
+        } else if (needToExcludeIrrelevantFiles(fileParameters)) {
+            fileParameters.getFileList().removeIf(file -> !isaFileAccordingToConditions(fileParameters, file));
+        }
+        next(filePath, fileParameters);
+    }
+
+    private void walkingOfDirectoryByRecursion(File filePath, FileParameters fileParameters) {
+        if (filePath.isDirectory()) {
+            for (Object fileObj : ArrayUtils.nullToEmpty(filePath.listFiles())) {
```

Comment:
There is overloaded method in ArrayUtils::nullToEmpty that can return parametrized value instead of Object